packages/sync-actions/src/utils/create-build-actions.js

```diff
@@ -35,9 +35,10 @@ function injectMissingPriceIds(nextVariants, previousVariants) {
     if (!prices) return restOfVariant
     const oldVariant = previousVariants.find(
       previousVariant =>
-        previousVariant.id === newVariant.id ||
-        previousVariant.key === newVariant.key ||
-        previousVariant.sku === newVariant.sku
+        (!isNil(previousVariant.id) && previousVariant.id === newVariant.id) ||
+        (!isNil(previousVariant.key) &&
+          previousVariant.key === newVariant.key) ||
+        (!isNil(previousVariant.sku) && previousVariant.sku === newVariant.sku)
```

Comment:
Why not simply like below?
```
(previousVariant.id && previousVariant.id === newVariant.id) ||
(previousVariant.key && previousVariant.key === newVariant.key) ||
(previousVariant.sku && previousVariant.sku === newVariant.sku)
```

Reply:
This was mainly because of id - if we start indexing from zero, we would not match variants with `id` equal to `0`